website/src/types/mpe.ts

```diff
@@ -0,0 +1,26 @@
+import { ModuleTitle, ModuleCode } from './modules';
+
+export interface EssentialMajor {
+  type: '01';
+}
+
+export interface EssentialSecondMajor {
+  type: '02';
+}
+
+export interface Elective {
+  type: '03';
+}
+
+export interface UnrestrictedElective {
+  type: '04';
+}
+
+export type ModuleType = EssentialMajor | EssentialSecondMajor | Elective | UnrestrictedElective;
+
+export type MpePreference = {
+  moduleTitle?: ModuleTitle;
+  moduleCode: ModuleCode;
+  moduleType: ModuleType;
```

Comment:
I think a better way to model this data is to represent type as a union of constant strings 
```suggestion
  moduleType: 'essential' | 'essentialSecondMajor' | 'elective' | 'unrestrictedElective';
```

You would only represent the types as objects themselves if you want to store additional info on type, but we don't need to in our case. You can then store data associated with each ModuleType in a separate Record 

``` ts 
interface ModuleTypeInfo {
  label: string;
  code: string;
}

const MODULE_TYPES: Record<MpePreference['moduleType']> = {
  essential: {
    label: 'Essential Major',
    code: '04',
  },
  // ...
};
```